reports-scheduler/src/main/kotlin/com/amazon/opendistroforelasticsearch/reportsscheduler/resthandler/ReportStatsRestHandler.kt

```diff
@@ -0,0 +1,87 @@
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ *
+ */
+package com.amazon.opendistroforelasticsearch.reportsscheduler.resthandler
+
+import com.amazon.opendistroforelasticsearch.reportsscheduler.ReportsSchedulerPlugin.Companion.BASE_REPORTS_URI
+import com.amazon.opendistroforelasticsearch.reportsscheduler.action.ReportDefinitionActions
+import com.amazon.opendistroforelasticsearch.reportsscheduler.model.RestTag.STATS_START_TIME
+import com.amazon.opendistroforelasticsearch.reportsscheduler.model.RestTag.STATS_END_TIME
+import com.amazon.opendistroforelasticsearch.reportsscheduler.metrics.Metrics
+import org.elasticsearch.client.node.NodeClient
+import org.elasticsearch.rest.BaseRestHandler
+import org.elasticsearch.rest.BaseRestHandler.RestChannelConsumer
+import org.elasticsearch.rest.BytesRestResponse
+import org.elasticsearch.rest.RestHandler.Route
+import org.elasticsearch.rest.RestRequest
+import org.elasticsearch.rest.RestRequest.Method.GET
+import org.elasticsearch.rest.RestStatus
+
+/**
+ * Rest handler for getting reporting backend stats
+ * This handler uses [ReportDefinitionActions]. TODO: change this
+ */
+internal class ReportStatsRestHandler : BaseRestHandler() {
+    companion object {
+        private const val REPORT_STATS_ACTION = "report_definition_stats"
+        private const val REPORT_STATS_URL = "$BASE_REPORTS_URI/stats"
+        private const val REPORT_TEST_URL = "$BASE_REPORTS_URI/test"
+        public const val COUNT = 4
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    override fun getName(): String {
+        return REPORT_STATS_ACTION
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    override fun routes(): List<Route> {
+        return listOf(
+            /**
+             * Get reporting backend stats
+             * Request URL: GET REPORT_STATS_URL
+             * Request body: Ref [com.amazon.opendistroforelasticsearch.reportsscheduler.model.GetReportDefinitionRequest]
+             * Response body: Ref [com.amazon.opendistroforelasticsearch.reportsscheduler.model.GetReportDefinitionResponse]
+             */
+            Route(GET, "$REPORT_STATS_URL"),
+            Route(GET, "$REPORT_TEST_URL")
+        )
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    override fun responseParams(): Set<String> {
+        return setOf(STATS_START_TIME, STATS_END_TIME)
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    override fun prepareRequest(request: RestRequest, client: NodeClient): RestChannelConsumer {
+        return when (request.method()) {
+            GET -> RestChannelConsumer {
+                it.sendResponse(BytesRestResponse(RestStatus.OK, Metrics.getInstance().collectToFlattenedJSON()))
```

Comment:
by making it not calling transport action, user permission is not checked. is this behavior expected?

Reply:
I did not see a requirement to check user permission for this API. Do you see a scenario where exposing this to all users be a problem?

Reply:
Will evaluate if calling transport action in another PR. Leaving this for now.